app/forms/state_file/az_retirement_income_form.rb

```diff
@@ -0,0 +1,33 @@
+module StateFile
+  class AzRetirementIncomeForm < QuestionsForm
+    set_attributes_for :intake,
+                       :received_military_retirement_payment,
+                       :received_military_retirement_payment_amount,
+                       :primary_received_pension,
+                       :primary_received_pension_amount,
+                       :spouse_received_pension,
+                       :spouse_received_pension_amount
+
+    validates :received_military_retirement_payment_amount, presence: true, numericality: true, allow_blank: false, if: -> { received_military_retirement_payment == "yes" }
+    validates :primary_received_pension_amount, presence: true, numericality: true, allow_blank: false, if: -> { primary_received_pension == "yes" }
+    validates :spouse_received_pension_amount, presence: true, numericality: true, allow_blank: false, if: -> { spouse_received_pension == "yes" }
+    validate :below_1040_amount, if: -> { [received_military_retirement_payment_amount, primary_received_pension_amount, spouse_received_pension_amount].any?(&:present?) }
+
+    def save
+      @intake.update(attributes_for(:intake))
+    end
+
+    private
+
+    def below_1040_amount
+      amount_limit = @intake.direct_file_data.fed_taxable_pensions
```

Comment:
This is interesting to me, because I feel like this is probably something that doesn't change per year. Is it required to be loaded at validation time?

Reply:
i think it's the amount the client received in pensions that year? so i would expect it to change per year. i'm not sure what you mean by "loaded" at validation time? is there a way to pre-load it?

Reply:
It currently happens at the moment you call `valid?` or `save?` and looks up the value. If it didn’t happen there it could happen instead on class load. But it’s somewhat moot because of the errors being established in this method. Mostly I was thinking whether this case is special enough to be broken out or if there’s a way we could do it with regular validators. I think setting the error message on each makes it special enough though.